sdk/cosmos/azure_data_cosmos/src/clients/container_client.rs

```diff
@@ -60,15 +60,16 @@ impl ContainerClient {
     /// ```
     pub async fn read(
         &self,
-
-        #[allow(unused_variables)]
-        // REASON: This is a documented public API so prefixing with '_' is undesirable.
-        options: Option<ReadContainerOptions>,
+        options: Option<ReadContainerOptions<'_>>,
     ) -> azure_core::Result<Response<ContainerProperties>> {
         let url = self.pipeline.url(&self.link);
         let mut req = Request::new(url, Method::Get);
         self.pipeline
-            .send(Context::new(), &mut req, self.link.clone())
+            .send(
+                options.map(|o| o.method_options.context),
```

Comment:
This will work for now, but once we figure out the final tracing story, you may have to change it to something like https://gist.github.com/heaths/3aea54ccac2e404d22e110b2400e9200

Reply:
Yeah. I was also considering using `impl AsClientMethodOptions` in the `self.pipeline.send` helper (which is internal, so I don't mind using an `impl` type there) to reduce boilerplate, but I decided to wait on that and see what we need later. This is all internal code, so refactoring is straightforward.